src/test/java/uk/gov/pay/directdebit/payments/dao/PaymentViewDaoIT.java

```diff
@@ -434,32 +402,23 @@ public void shouldReturn6Records_whenSearchingByFailedExternalState() {
                 .withMandateFixture(mandateFixture)
                 .withState(PaymentState.CREATED)
                 .insert(testContext.getJdbi());
-        for (int i = 0; i < 6; i++) {
-            if (i % 3 == 0) {
-                aPaymentFixture()
-                        .withMandateFixture(mandateFixture)
-                        .withState(PaymentState.CANCELLED)
-                        .insert(testContext.getJdbi());
-                continue;
-            }
-            if (i % 2 == 0) {
-                aPaymentFixture()
-                        .withMandateFixture(mandateFixture)
-                        .withState(PaymentState.EXPIRED)
-                        .insert(testContext.getJdbi());
-                continue;
-            }
+        
+        List.of(1,2).forEach(i -> {
```

Comment:
Perhaps `IntStream.rangeClosed(1, 2).forEach` might be better if its only needed to generate a stream of ints/drive the `forEach` as I was looking for where the `List` is being referenced or used later on.

Reply:
I get what you mean but it its an improvement on `for (int i = 0; i < 6; i++)`...`if (i % 3 == 0)`...`if (i % 2 == 0)`

Reply:
oh yeah certainly, I don't mind it as-is, just thought I mention it.